scientific_library/tvb/analyzers/wavelet.py

```diff
@@ -41,208 +41,147 @@
 
 import numpy
 import scipy.signal as signal
-import tvb.datatypes.time_series as time_series
 import tvb.datatypes.spectral as spectral
+from tvb.basic.logger.builder import get_logger
 from tvb.basic.neotraits.api import HasTraits, Attr, Range, Float, narray_describe
 from tvb.simulator.common import iround
 
-
 SUPPORTED_WAVELET_FUNCTIONS = ("morlet",)
 
+log = get_logger(__name__)
 
+"""
+A module for calculating the wavelet transform of a TimeSeries object of TVB
+and returning a WaveletSpectrum object. The sampling period and frequency
+range of the result can be specified. The mother wavelet can also be
+specified... (So far, only Morlet.)
+
+References:
+    .. [TBetal_1996] C. Tallon-Baudry et al, *Stimulus Specificity of
+        Phase-Locked and Non-Phase-Locked 40 Hz Visual Responses in Human.*,
+        J Neurosci 16(13):4240-4249, 1996.
+
+    .. [Mallat_1999] S. Mallat, *A wavelet tour of signal processing.*,
+        book, Academic Press, 1999.
+"""
 
-class ContinuousWaveletTransform(HasTraits):
+
+def evaluate(time_series, view_model):
```

Comment:
Exactly this form is not good (`view_model` id defined at a higher level but used here => circular dependency)

Reply:
So practically instead of sending the view model, I should sent just the needed parameters from the view model?